proxy-robots-module/src/main/java/orwell/proxy/Tank.java

```diff
@@ -0,0 +1,331 @@
+    private final Register.Builder registerBuilder = Register
+            .newBuilder();
+    private Input currentControllerInput;
+    private Registered serverGameRegistered;
+    private final NXTInfo nxtInfo;
+    private final MessageFramework mfTank;
+    private final ICamera camera;
+    private Register register;
+    private String image;
+
+    private EnumRegistrationState registrationState = EnumRegistrationState.NOT_REGISTERED;
+    private EnumConnectionState connectionState = EnumConnectionState.NOT_CONNECTED;
+    private String teamName;
+
+    private final TankDeltaState tankDeltaState = new TankDeltaState();
+
+    public Tank(String bluetoothName, String bluetoothID, ICamera camera,
+                MessageFramework mf, String image) {
+        setBluetoothName(bluetoothName);
+        setBluetoothID(bluetoothID);
+        this.camera = camera;
+        nxtInfo = new NXTInfo(NXTCommFactory.BLUETOOTH, bluetoothName,
+                bluetoothID);
+        mfTank = mf;
+        mf.addMessageListener(this);
+        this.image = image;
+    }
+
+    public Tank(String bluetoothName, String bluetoothID, ICamera camera, String image) {
+        this(bluetoothName, bluetoothID, camera, new MessageFramework(), image);
+    }
+
+    @Override
+    public void buildRegister() {
+        registerBuilder.setTemporaryRobotId(routingID);
+        registerBuilder.setVideoUrl(camera.getUrl());
+        registerBuilder.setImage(image);
+        if ("" == image) {
+            logback.info("Image of tank " + routingID + " is empty. "
+                    + "This will probably be an issue for the serverGame");
+        }
+        register = registerBuilder.build();
+    }
+
+    private Register getRegister() {
+        if (null == register) {
+            buildRegister();
+        }
+        return register;
+    }
+
+    @Override
+    public byte[] getRegisterBytes() {
+        Register register = getRegister();
+        if (null != register)
+            return register.toByteArray();
+        else
+            return null;
+    }
+
+    @Override
+    public Input getControllerInput() {
+        return currentControllerInput;
+    }
+
+    @Override
+    public void setControllerInput(byte[] inputMessage) {
+        try {
+            this.currentControllerInput = Input.parseFrom(inputMessage);
+            if (currentControllerInput.hasMove()) {
+                String payloadMove = "input move ";
+                payloadMove += currentControllerInput.getMove().getLeft() + " "
+                        + currentControllerInput.getMove().getRight();
+                UnitMessage msg = new UnitMessage(UnitMessageType.Command,
+                        payloadMove);
+                mfTank.SendMessage(msg);
+            }
+            if (currentControllerInput.hasFire()
+                    && (currentControllerInput.getFire().getWeapon1() || currentControllerInput
+                    .getFire().getWeapon2())) {
```

Comment:
It seems that you are doing a strange test here. Maybe parentheses are missing around the or part.